pkg/deploy/generator/scripts/gatewayVMSS.sh

```diff
@@ -316,7 +316,20 @@ echo "Download \$COMPONENT credentials"
 
 TEMP_DIR=\$(mktemp -d)
 export AZURE_CONFIG_DIR=\$(mktemp -d)
-az login -i --allow-no-subscriptions
+
+echo "Logging into Azure..."
+RETRIES=3
+while [ "\$RETRIES" -gt 0 ]; do
+    if az login -i --allow-no-subscriptions
+    then
+        echo "az login successful"
+        break
+    else
+        echo "az login failed. Retrying..."
+        let RETRIES-=1
+        sleep 5
+    fi
+done
```

Comment:
No real difference in the logical outcome of each loop type. I will say though that I was actually going for consistency with how we use `az login -i` everywhere else in ARO, because this loop-style is what we use in all of our pipeline scripts, [example 1](https://dev.azure.com/msazure/AzureRedHatOpenShift/_git/ARO.Pipelines?path=/ev2/Aks-Deployment/ServiceGroupRoot/bin/register.sh&version=GBmaster&line=14&lineEnd=24&lineStartColumn=1&lineEndColumn=5&lineStyle=plain&_a=contents), [2](https://dev.azure.com/msazure/AzureRedHatOpenShift/_git/ARO.Pipelines?path=/ev2/Bootstrapper/ServiceGroupRoot/bin/bootstrapper.sh&version=GBmaster&line=25&lineEnd=36&lineStartColumn=1&lineEndColumn=5&lineStyle=plain&_a=contents), [3](https://dev.azure.com/msazure/AzureRedHatOpenShift/_git/ARO.Pipelines?path=/ev2/Deployment/ServiceGroupRoot/bin/deploy.sh&version=GBmaster&line=14&lineEnd=24&lineStartColumn=1&lineEndColumn=5&lineStyle=plain&_a=contents), [4](https://dev.azure.com/msazure/AzureRedHatOpenShift/_git/ARO.Pipelines?path=/ev2/Hive-Deployment/ServiceGroupRoot/bin/deploy-hive.sh&version=GBmaster&line=17&lineEnd=27&lineStartColumn=5&lineEndColumn=9&lineStyle=plain&_a=contents), etc.